packages/instant/src/util/script_environment.ts

```diff
@@ -0,0 +1,60 @@
+import * as _ from 'lodash';
+
+import { ENVIRONMENTS_TO_DOMAINS } from '../constants';
+import { Environment } from '../types';
+
+export type EnvironmentToDomain = { [key in Environment]: string[] };
+
+export const scriptEnvironment = {
+    // Attempts to figure out full URL of where this JS is being served from
+    getFullScriptLocation: (): string | null => {
+        const currentScript = document.currentScript;
+
+        if (!currentScript) {
+            return null;
+        }
+
+        const scriptSrc = currentScript.getAttribute('src');
+        if (!scriptSrc) {
+            return null;
+        }
+
+        try {
+            const url = new URL(scriptSrc, window.location.href);
+            return url.href;
+        } catch {
+            return null;
+        }
+    },
+    urlToEnvironment: (url: string): Environment | undefined => {
+        const allEnvironments = [
+            Environment.Production,
+            Environment.Staging,
+            Environment.Dogfood,
+            Environment.Development,
+        ];
+        for (const curEnvironment of allEnvironments) {
+            const domains = ENVIRONMENTS_TO_DOMAINS[curEnvironment];
+            if (_.some(domains, (domain: string) => url.includes(domain))) {
+                return curEnvironment;
+            }
+        }
+        return undefined;
+    },
+    getEnvironment: (defaltValue = Environment.Production): Environment => {
```

Comment:
On second look, if we can just identify the environment in the build process (i.e. https://github.com/0xProject/0x-monorepo/pull/1272 ), we could use that instead of doing all of this logic to identify the environment